### PR TITLE
Change `dependency-build`goal from phase `prepare-package` to `process-resources`. Fix #81

### LINKED ISSUE #81: dependency-build should be run before lint
**BUG REPORT** :
**Environment (plugin version, maven version, OS, ...)**: 5.3

When project has dependency in Chart.yaml below configuration doesn't work:
```
<plugin>
    <groupId>com.kiwigrid</groupId>
    <artifactId>helm-maven-plugin</artifactId>
    <version>5.3</version>
    <executions>
        <execution>
            <id>helm</id>
            <goals>
                <goal>init</goal>
                <goal>dependency-build</goal>
                <goal>lint</goal>
                <goal>package</goal>
                <goal>upload</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
    </configuration>
</plugin>

``` 
because goal `dependency-build` is run after `lint`, so dependency is not resolved.

`dependency-build` is bind to `prepare-package` phase and `lint` is bind to `test` phase. 

`dependency-build` should be bind to some earlier phase, maybe `compile`

This configuration is workaround:

```
<plugin>
    <groupId>com.kiwigrid</groupId>
    <artifactId>helm-maven-plugin</artifactId>
    <version>5.3</version>
    <executions>
        <execution>
            <id>helm-init</id>
            <phase>initialize</phase>
            <goals>
                <goal>init</goal>
            </goals>
        </execution>
        <execution>
            <id>helm-update</id>
            <phase>compile</phase>
            <goals>
                <goal>dependency-build</goal>
            </goals>
        </execution>
        <execution>
            <id>helm</id>
            <goals>
                <goal>lint</goal>
                <goal>package</goal>
                <goal>upload</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
    </configuration>
</plugin>
```

**Packaging with the Helm Lifecycle**
I suspect that Packaging with the Helm Lifecycle also will not work for chart with dependency.

We have in plugin life cycle configuration:
```
<lifecycle>
    <id>default</id>
    <phases>
        <initialize>com.kiwigrid:helm-maven-plugin:init</initialize>
        <test>com.kiwigrid:helm-maven-plugin:lint</test>
        <package>com.kiwigrid:helm-maven-plugin:package</package>
        <deploy>com.kiwigrid:helm-maven-plugin:upload</deploy>
    </phases>
</lifecycle>
```

so goal `dependency-build` is not bind to any phase

**What you expected to happen**:
build pass

### PR DESCRIPTION
**Pull request by slawekjaranowski**
Examples for #81 

- lint-with-dependency - use current bindings
- lint-with-dependency-workaround - show that chart is ok if `dependency-build` is executed before `lint`
- lint-with-dependency-extension - the same chart - using helm packaging

As we see `lint-with-dependency` and `lint-with-dependency-extension` will fail.

**TODO to fix** 
- change binding of  dependency-build
- remove int-with-dependency-workaround

### CODE REVIEW

**Comment by sschnabe** [comment]
@slawekjaranowski i've changed the phase and adjusted the it-tests. Ok for you?

**Comment by slawekjaranowski** [comment]
great, it is ok for me.

File: src/it/extension/src/main/helm/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v2
+name: app
+version: 0.0.1
+
+dependencies:
+  - name: mysqldump
+    version: 5.0.0
+    repository: "@kokuwaiuo"
```
Comment (axdotl) [inline src/it/extension/src/main/helm/Chart.yaml:8]:
```suggestion
    repository: "@kokuwaio"
```

File: src/it/lint-with-dependency/pom.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<chartDirectory>${project.basedir}/src/main/helm</chartDirectory>
+					<chartVersion>${project.version}</chartVersion>
+					<addDefaultRepo>false</addDefaultRepo>
+					<helmExtraRepos>
+						<helmExtraRepo>
+							<name>kokuwaiuo</name>
```
Comment (axdotl) [inline src/it/lint-with-dependency/pom.xml:34]:
```suggestion
							<name>kokuwaio</name>
```

File: src/it/lint-with-dependency/src/main/helm/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v2
+name: app
+version: 0.0.1
+
+dependencies:
+  - name: mysqldump
+    version: 5.0.0
+    repository: "@kokuwaiuo"
```
Comment (axdotl) [inline src/it/lint-with-dependency/src/main/helm/Chart.yaml:8]:
```suggestion
    repository: "@kokuwaio"
```

**Review by rpahli — APPROVED** [review]
LGTM

**Review by fabian-schlegel — APPROVED** [review]

**Pull request merged**